lib/client/Client.js

```diff
@@ -336,23 +336,47 @@ export default class Client {
 
   /**
    * Destroy the {@link Mailbox} related to the given _id_ on the server.<br />
-   * This will issue a {@link Client#setMailboxes} JMAP request under the hoods, passing the correct options.
+   * This will issue a {@link Client#destroyMailboxes} request under the hoods, passing _[id]_ option.
    *
    * @param id {String} The id of the mailbox to destroy.
    *
    * @return {Promise} A {@link Promise} that eventually resolves to nothing if the {@link Mailbox} was destroyed successfully.
    *
-   * @see http://jmap.io/spec.html#destroying-mailboxes
    */
   destroyMailbox(id) {
     Utils.assertRequiredParameterIsPresent(id, 'id');
 
-    return this.setMailboxes({destroy: [id]})
-      .then(response => {
-        if (response.destroyed.indexOf(id) < 0) {
-          throw new Error('Failed to destroy ' + id + ', the reason is: ' + (response.notDestroyed[id] || 'missing'));
-        }
-      });
+    return this.destroyMailboxes([id]);
+  }
+
+  /**
+   * Destroy multiple {@link Mailbox}es specified to the given _ids_ on the server.<br />
+   * This will issue a {@link Client#setMailboxes} JMAP request under the hoods, passing the correct options.
+   *
+   * @param ids {String[]} An array IDs of the mailboxes to destroy. These IDs must be in the right order: Destroy X comes before destroy Y if X is a descendent of Y.
+   *
+   * @return {Promise} A {@link Promise} that eventually resolves to nothing if all {@link Mailbox}es were destroyed successfully.
+   *
+   * @see http://jmap.io/spec.html#destroying-mailboxes
+   */
+  destroyMailboxes(ids) {
+    Utils.assertRequiredParameterIsArrayWithMinimumLength(ids, 'ids', 1);
+
+    return this.setMailboxes({
+      destroy: ids
+    }).then(response => {
+      let notDestroyedIds = Object.keys(response.notDestroyed);
+
+      if (notDestroyedIds.length > 0) {
```

Comment:
You finally did as you wished... I don't want to block this forever so I'll approve, but you're aware that as soon as we want to update the UI in OpenPaas we'll need to use setMailboxes and this helper method will become useless...